### PR TITLE
dependency hell

### PR DESCRIPTION
**Pull request by SansPapyrus683**
_Place an "x" in the corresponding checkbox if it is done or does not apply to this pull request._

- [x] I have tested my code.
- [x] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [x] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions).
  - I understand that if it is clear that I have not attempted to follow these conventions, my PR will be closed.
  - If changes are requested, I will re-request a review after addressing them.
- [x] I have linked this PR to any issues that it closes.

don't ask why the commits are under some other guy

### CODE REVIEW

**Comment by SansPapyrus683** [comment]
if this builds successfully i will literally breathe once

**Comment by SansPapyrus683** [comment]
NOOOOOOO

**Comment by SansPapyrus683** [comment]
WHY IS ESLINT WHINING NOW

**Comment by danielzsh** [comment]
Wait why all the changes

**Comment by danielzsh** [comment]
oh lol ive been ignoring them this whole time but honestly if dependabot doesn't complain the "deprecation" doesn't matter

**Comment by SansPapyrus683** [comment]
it matters to me :sob: 

image

File: gatsby-node.ts [hunk 1]
```diff
@@ -202,7 +202,7 @@ exports.createPages = async ({ graphql, actions, reporter }) => {
       });
   });
   const result = await graphql(`
-    query {
+    query stuff {
```
Comment (danielzsh) [inline gatsby-node.ts:205]: wait why

**Comment by danielzsh** [comment]
wait wth this does not make sense
error failed to index to Algolia Invalid value "undefined" for header "x-algolia-api-key"


  TypeError:Invalid value "undefined" for header "x-algolia-api-key"

File: gatsby-node.ts (hunk 1 above)
Comment (SansPapyrus683) [reply gatsby-node.ts:205]: gatsby was complaining about "anonymous queries"

**Comment by SansPapyrus683** [comment]
> did you set the algolia secret in your fork? Also make sure you didn't set ALGOLIA_APP_ID...index to Algolia should be skipped

i have no idea what this means lol

**Comment by danielzsh** [comment]
Does yarn build work locally?

File: gatsby-node.ts (hunk 1 above)
Comment (danielzsh) [reply gatsby-node.ts:205]: odd...maybe put a more descriptive name then 😅

Comment (SansPapyrus683) [reply gatsby-node.ts:205]: will fix when i get the rest of the stuff working lmao

**Comment by SansPapyrus683** [comment]
my computer literally runs out of ram before it finishes

**Comment by SansPapyrus683** [comment]
```
Command failed with exit code 127: yarn run build-storybook --output-dir /tmp/chromatic--2899-qh1vuH6gVwO9
/bin/sh: 1: build-storybook: not found
error Command failed with exit code 127.
```
WHAT

**Comment by SansPapyrus683** [comment]
image

image

oh no

**Comment by SansPapyrus683** [comment]
WE TAKE THESE
WE TAKE THESE

**Comment by SansPapyrus683** [comment]
ALRIGHT ALRIGHT

**Comment by SansPapyrus683** [comment]
it's actually over

**Comment by SansPapyrus683** [comment]
image

WHHAT IS THIS

**Comment by SansPapyrus683** [comment]
oh wow NOW it works

**Comment by SansPapyrus683** [comment]
of frickin' course it was.

**Comment by SansPapyrus683** [comment]
actual cancer

**Comment by SansPapyrus683** [comment]
gotta love it when something COMPLETELY UNRELATED to the thing i'm changing breaks

**Comment by SansPapyrus683** [comment]
WHAT

**Comment by danielzsh** [comment]
hes still going at it 😭

**Comment by envyaims** [comment]
more persistent than a pst

**Comment by SansPapyrus683** [comment]
daniel i'd really appreciate it if you could help me out here w the algolia stuff

**Comment by danielzsh** [comment]
image
this diff is mental

wait so does `yarn build` work locally or no

**Comment by SansPapyrus683** [comment]
i told u my computer literally runs out of ram

**Comment by danielzsh** [comment]
I feel like this pr may be a bit beyond saving...what were the original packages that needed upgrading? (btw i don't think you upgraded react, but note react 18 kinda breaks our codebase)

**Comment by SansPapyrus683** [comment]
> react 18 kinda breaks our codebase

💀

**Comment by SansPapyrus683** [comment]
> I feel like this pr may be a bit beyond saving...what were the original packages that needed upgrading? (btw i don't think you upgraded react, but note react 18 kinda breaks our codebase)

i was just tryna upgrade as many dependencies as possible

**Comment by danielzsh** [comment]
maybe just close this pr and try going one by one instead?

**Comment by SansPapyrus683** [comment]
great idea
i'll try that over spring break, it's currently finals season

**Pull request closed without merging**